pkg/blockstorage/vmware/vmware.go

```diff
@@ -358,14 +362,52 @@ func (p *FcdProvider) setTagsVolume(ctx context.Context, volume *blockstorage.Vo
 	return nil
 }
 
+func (p *FcdProvider) setTagsSnapshot(ctx context.Context, snapshot *blockstorage.Snapshot, tags map[string]string) error {
+	if snapshot == nil {
+		return errors.New("Empty snapshot")
+	}
+	_, snapshotID, err := SplitSnapshotFullID(snapshot.ID)
+	if err != nil {
+		return errors.Wrap(err, "Cannot infer volume ID from full snapshot ID")
+	}
+	for cat, val := range tags {
+		err := p.Gom.AttachTag(ctx, vimID(snapshotID), cat, val)
+		if err != nil {
+			return errors.Wrap(err, "Failed to update metadata")
+		}
+	}
+	return nil
+}
+
 // VolumesList is part of blockstorage.Provider
 func (p *FcdProvider) VolumesList(ctx context.Context, tags map[string]string, zone string) ([]*blockstorage.Volume, error) {
 	return nil, errors.New("Not implemented")
 }
 
 // SnapshotsList is part of blockstorage.Provider
 func (p *FcdProvider) SnapshotsList(ctx context.Context, tags map[string]string) ([]*blockstorage.Snapshot, error) {
-	return nil, errors.New("Not implemented")
+	var snapshotIDs []types.ID
+	var err error
+	if len(tags) == 1 {
+		for k, v := range tags {
```

Comment:
These two lines look suspicious to me. If there is now error in the `if` condition, then probably there is no need in `for`.